### PR TITLE
Add dev.Dockerfile for FwHeadless

### LINKED ISSUE #1190: Create dev.Dockerfile for FwHeadless
Currently FwHeadless only has a Dockerfile, so every time you change any of its code it rebuilds the Docker image. We should change it to use a dev.Dockerfile similar to LexBoxApi, so that we can run `dotnet watch` in the container and have Skaffold sync files to it, which will save a lot of time when we're iterating on a new feature.

### PR DESCRIPTION
**Pull request by rmunn**
Fix #1190.

This creates a Dockerfile for fw-headless running `dotnet watch` to make it easier to do development inside the container.

### CODE REVIEW

**Review by hahn-kev — COMMENTED** [review]
putting this on hold until #1204 is merged in and we'll setup the sync there

File: backend/FwHeadless/dev.Dockerfile
Comment (rmunn) [reply backend/FwHeadless/dev.Dockerfile]: Done in commit c64f560f.

**Review by hahn-kev — APPROVED** [review]

**Pull request merged**
